ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/PreventLocalNodeConnection.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.p2p;
+
+public class PreventLocalNodeConnection extends RuntimeException {
```

Comment:
This probably should be more about what went wrong rather than why we don't allow it.  So `ConnectingToLocalPeerException` maybe?  Also good to end exception names with `Exception` (though we break that convention in a few places).

Reply:
Done.